src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -253,6 +274,25 @@ public DataServiceContext(Uri serviceRoot, ODataProtocolVersion maxProtocolVersi
         {
         }
 
+        /// <summary>Initializes a new instance of the <see cref="Microsoft.OData.Client.DataServiceContext" /> class with the specified <paramref name="serviceRoot" /> and targeting the specific <paramref name="maxProtocolVersion" />.</summary>
+        /// <param name="serviceRoot">An absolute URI that identifies the root of a data service.</param>
+        /// <param name="maxProtocolVersion">A <see cref="Microsoft.OData.Client.ODataProtocolVersion" /> value that is the maximum protocol version that the client understands.</param>
+        /// <param name="httpClientHandlerProvider" > Provides an instance of<see cref="HttpClientHandler"/> to use when making a request
+        /// under the <see cref="HttpRequestTransportMode.HttpClient"/>. If no <see cref="IHttpClientHandlerProvider"/> is specified, a new <see cref="HttpClientHandler"/>
+        /// will created for each request.</param>
+        /// <remarks>
+        /// The library expects the Uri to point to the root of a data service,
+        /// but does not issue a request to validate it does indeed identify the root of a service.
+        /// If the Uri does not identify the root of the service, the behavior of the client library is undefined.
+        /// A Uri provided with a trailing slash is equivalent to one without such a trailing character.
+        /// With Silverlight, the <paramref name="serviceRoot"/> can be a relative Uri
+        /// that will be combined with System.Windows.Browser.HtmlPage.Document.DocumentUri.
+        /// </remarks>
+        public DataServiceContext(Uri serviceRoot, ODataProtocolVersion maxProtocolVersion, IHttpClientHandlerProvider httpClientHandlerProvider)
+            : this(serviceRoot, maxProtocolVersion, ClientEdmModelCache.GetModel(maxProtocolVersion), httpClientHandlerProvider: httpClientHandlerProvider)
+        {
+        }
+
         /// <summary>

```

Comment:
Given that the httpclienthandlerprovider is settable on the DataServiceContext, is there any reason we need to add constructor overloads for it?

Reply:
Not it's not necessary. Just that I started with the constructor argument then added the setter after the fact.

Since the `HttpRequestTransportMode` is set via property and not constructor, it does make sense to keep this on the property only.